src/EcalLumiSpecWScFi_geo.cpp

```diff
@@ -0,0 +1,201 @@
+                                                  modVol, Transform3D( RotationZYX( x_rot.z(), -90.0*degree, -90.0*degree), Position( mod_pos_x, mod_pos_y, mod_pos_z ) ) );
+
+                                  modPV.addPhysVolID( "sector", sector_id ).addPhysVolID( "module", mod_id);
+                                  mod_id++;
+                          }//imod-loop close
+                  }//if-close
+                  else{// 1 : || to y-axis
+                          for(int imod=0; imod< nmod_perlayer; imod++){
+
+                                  double        mod_pos_z       = ( x_pos.z() - 0.0*cm ) - layer_pos0 - ilay*modSize.y();
+                                  double        mod_pos_x       = x_pos.x() + mod_pos0 + imod*modSize.x();
+                                  double        mod_pos_y       = x_pos.y() + 0.0*mm;
+
+                                  PlacedVolume modPV = assembly.placeVolume(
+                                                  modVol, Transform3D( RotationZYX(x_rot.z(), x_rot.y(), -90.0*degree), Position( mod_pos_x, mod_pos_y, mod_pos_z ) ) );
+
+                                  modPV.addPhysVolID( "sector", sector_id ).addPhysVolID( "module", mod_id);
+                                  mod_id++;
+                          }//imod-loop close
+                  }//else-close
+
+          }//ilay-loop close
+  }// sectors
+
+  // Place assembly into mother volume.  Assembly is centered at origin
+  PlacedVolume detPV = motherVol.placeVolume( assembly, Position(0.0, 0.0, 0.0) );
+  detPV.addPhysVolID("system", det_ID);
+  // Connect to system ID
+  det.setPlacement(detPV);
+
+  return det;
+} //Driver class close
+
+static tuple<Volume, Position> build_specScFiCAL_module( const Detector& description, const xml::Component& mod_x, SensitiveDetector& sens){
+
+        //--------------------Module Setup---------------------------------------------------------------------
+        double sx = mod_x.attr<double>(_Unicode(sizex));
+        double sy = mod_x.attr<double>(_Unicode(sizey));
+        double sz = mod_x.attr<double>(_Unicode(sizez));
+
+        Position modSize(sx, sy, sz);
+
+        Box    modShape( modSize.x()/2.0, modSize.y()/2.0, modSize.z()/2.0 );
+        auto   modMat = description.material(mod_x.attr<std::string>(_Unicode(material)));
+        Volume modVol("module_vol", modShape, modMat);
+
+        modVol.setVisAttributes(description.visAttributes(mod_x.attr<std::string>(_Unicode(vis))));
+        //----------------------------Scintillating fibers -----------------------------------------------------------
+
+        //detectors will be placed in module within square shaped sub-modules inside the modules.
+        /*
+           | + + + + + | x (->) even
+           |  + + + +  | y (up) odd
+           | + + + + + | even
+           |  + + + +  | odd
+           fsx, fsy = space between center of two fibers or submods along x, y in fig.
+           foff = space between two modules
+           */
```

Comment:
If this comment is about the fibre layout within a module it doesn't match the current implementation, if it is about x,y orientation of layers it is not entirely clear and should be moved to where the layers are defined.